frontend/src/components/header.js

```diff
@@ -10,17 +10,29 @@ import { Route, NavLink } from "react-router-dom";
  * @param {*} props
  * @returns
  */
-export function BootstrapNavLink(props) {
+function BootstrapNavItem(props) {
     return (
-        <Nav.Link as={NavLink} activeClassName="text-dark" {...props}>
-            {props.children}
-        </Nav.Link>
+        <Nav.Item>
+            <Nav.Link as={NavLink} {...props}>
+                {props.children}
+            </Nav.Link>
+        </Nav.Item>
     );
 }
-BootstrapNavLink.propTypes = {
+BootstrapNavItem.propTypes = {
     to: PropTypes.string,
+    active: PropTypes.bool,
 };
 
+/**
+ * Returns the top level route for a full path ("/tapp/sessions" -> "/tapp")
+ * @param {string} pathname
+ */
+function getRoute(pathname) {
```

Comment:
You shouldn't be doing route parsing manually...`react-router-dom` should have all these functions built in.

Reply:
i am using react-router-dom to get the path, getRoute just clips off anything after the top level route (as described in the docstring)

Reply:
Yes, but they should have facility for deep linking. If you set things up as subroutes, it should all work without any parsing.

Reply:
https://stackoverflow.com/questions/41474134/nested-routes-with-react-router-v4-v5

Reply:
Are you still working on this? I don't see it changed yet.